### PR TITLE
Fixup docker-compose to run local ocluster-worker.

### PR DESCRIPTION
**Pull request by tmcgilchrist**
Removes the linux/amd64 requirement, we publish arm64 builds of scheduler and worker so this should continue to work on Linux and MacOS (x86 and Arm64).

### CODE REVIEW

**Comment by tmcgilchrist** [comment]
The root Dockerfile is for the deployed service that runs on https://images.ci.ocaml.org, which gets checked by the deployability check. While the other docker file is for running an  ocluster-worker inside a docker container. We need this latter one to build a Docker in Docker image.

I haven't validated this on M1 macs, could you check @benmandrew or @novemberkilo ?

**Comment by benmandrew** [comment]
The issue I mentioned is fixed in #228; this PR can be merged with the changes detailed above (except for docker-compose.yml line 38).

**Comment by tmcgilchrist** [comment]
@benmandrew I've left a comment on that line to help with debug. Could you re-test to confirm everything starts up and a linux-arm64 build works?

File: docker-compose.yml [hunk 1]
```diff
@@ -30,15 +30,23 @@ services:
...
+      context: .
+    platform: linux
     command:
       - --connect=/capnp-secrets/pool-linux-x86_64.cap
+      # - --connect=/capnp-secrets/pool-linux-arm64.cap  # NOTE replace with arm64 if running on macos/arm64
       - --name=local
       - --allow-push=ocurrentbuilder/staging,ocurrent/opam-staging
       - --capacity=1
       - --state-dir=/var/lib/ocluster
+      - --store=rsync:/var/cache/obuilder
```
Comment (benmandrew) [inline docker-compose.yml:44]: It only works if I make the following change:
```suggestion
      - --obuilder-store=rsync:/var/cache/obuilder
```

Is the `--store` argument working for you?

(I am running `git submodule sync --recursive && git submodule update --recursive` to get the right submodules.)

**Review by benmandrew — CHANGES_REQUESTED** [review]

**Pull request merged**